### PR TITLE
Release notes 2.0.29

### PR DESCRIPTION
**Pull request by derekbekoe**
cc: @sptramer

### CODE REVIEW

**Comment by formulahendry** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs-ref-conceptual/release-notes-azure-cli.md](https://github.com/derekbekoe/azure-docs-cli-python/blob/az-2.0.29/docs-ref-conceptual/release-notes-azure-cli.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/cli/azure/release-notes-azure-cli?view=azure-cli-latest&branch=pr-en-us-683) (azure-cli-latest)<br/> [View](https://review.docs.microsoft.com/en-us/cli/azure/release-notes-azure-cli?view=azure-cli-2017-03-09-profile&branch=pr-en-us-683) (azure-cli-2017-03-09-profile) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C3%5C13%5C0c435b1f-40ec-55c2-6a92-4ae06d82d844%5CPullRequest%5C201803131728163680-683%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=YNIzzgJJDqbsHsh7b3L2Vwq%2FK7yQ5k7qx8Xlhho6h6s%3D&st=2018-03-13T17%3A26%3A07Z&se=2018-04-13T17%3A31%3A07Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs-ref-conceptual/release-notes-azure-cli.md [hunk 1]
```diff
@@ -14,6 +14,93 @@ ms.service: multiple
...
+* [BREAKING CHANGE] Renamed `advisor configuration get` to `advisor configuration list`
+* [BREAKING CHANGE] Renamed `advisor configuration set` to `advisor configuration update`
+* [BREAKING CHANGE] Removed `advisor recommendation generate` 
+* Added `--refresh` parameter to `advisor recommendation list`
+* Added `advisor recommendation show` command
+
+### Appservice
+
+* Deprecated `[webapp|functionapp] assign-identity`
+* webapp/functionapp: author managed identity commands `identity assign/show` [?]
```
Comment (sptramer) [inline docs-ref-conceptual/release-notes-azure-cli.md]: Not sure what this means.

File: docs-ref-conceptual/release-notes-azure-cli.md [hunk 2]
```diff
@@ -14,6 +14,93 @@ ms.service: multiple
...
+* Initial release
+
+### Extension
+
+* Added check to warn user if used distro is different then the one stored in package source file, as this may lead into errors
+
+### Interactive
+
+* Persist history across different sessions
+* Fixed history while in scope [#]
```
Comment (sptramer) [inline docs-ref-conceptual/release-notes-azure-cli.md]: Requires link to issue

File: docs-ref-conceptual/release-notes-azure-cli.md [hunk 3]
```diff
@@ -14,6 +14,93 @@ ms.service: multiple
...
+* Fixed progress meter for long running operations [#]
```
Comment (sptramer) [inline docs-ref-conceptual/release-notes-azure-cli.md]: Requires link to issue

File: docs-ref-conceptual/release-notes-azure-cli.md [hunk 4]
```diff
@@ -14,6 +14,93 @@ ms.service: multiple
...
+* Completions more robust to command table exceptions [?]
```
Comment (sptramer) [inline docs-ref-conceptual/release-notes-azure-cli.md]: Not sure what this means

File: docs-ref-conceptual/release-notes-azure-cli.md [hunk 5]
```diff
@@ -14,6 +14,93 @@ ms.service: multiple
...
+* Added `monitor autoscale rule` commands
+
+### Network
+
+* [BREAKING CHANGE] Removed `--tags` parameter from  `route-filter rule create`
+* Removed some erroneous default values for the following commands:
+  * `network express-route update`
+  * `network nsg rule update`
+  * `network public-ip update`
+  * `traffic manager profile update` [?]
```
Comment (sptramer) [inline docs-ref-conceptual/release-notes-azure-cli.md]: Not sure what command this actually is

File: docs-ref-conceptual/release-notes-azure-cli.md [hunk 6]
```diff
@@ -14,6 +14,93 @@ ms.service: multiple
...
+
+* Deprecated the `monitor autoscale-settings` commands
+* Added `monitor autoscale` commands
+* Added `monitor autoscale profile` commands
+* Added `monitor autoscale rule` commands
+
+### Network
+
+* [BREAKING CHANGE] Removed `--tags` parameter from  `route-filter rule create`
+* Removed some erroneous default values for the following commands:
```
Comment (sptramer) [inline docs-ref-conceptual/release-notes-azure-cli.md]: This was originally a "fixed" indicating it has an associated issue. Requires link(s)

**Comment by formulahendry** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs-ref-conceptual/release-notes-azure-cli.md](https://github.com/derekbekoe/azure-docs-cli-python/blob/az-2.0.29/docs-ref-conceptual/release-notes-azure-cli.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/cli/azure/release-notes-azure-cli?view=azure-cli-latest&branch=pr-en-us-683) (azure-cli-latest)<br/> [View](https://review.docs.microsoft.com/en-us/cli/azure/release-notes-azure-cli?view=azure-cli-2017-03-09-profile&branch=pr-en-us-683) (azure-cli-2017-03-09-profile) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C3%5C13%5C0c435b1f-40ec-55c2-6a92-4ae06d82d844%5CPullRequest%5C201803131822459399-683%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=MfOstWyDq8iMW9Bwg%2FoAXeM%2BIohOaRF11rl350nBwvo%3D&st=2018-03-13T18%3A21%3A13Z&se=2018-04-13T18%3A26%3A13Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs-ref-conceptual/release-notes-azure-cli.md [hunk 7]
```diff
@@ -14,6 +14,93 @@ ms.service: multiple
...
+* [PREVIEW] Changed to use the API 2017-12-01-preview
+
+### Service Bus
+
+* Initial release
+
+### Storage
+
+* Fixed issue with missing endpoint suffix in `storage blob copy` [#] [?]
+* Fixed issue where `storage blob [delete-batch|download-batch|upload-batch]` commands would throw an error [#]
```
Comment (sptramer) [inline docs-ref-conceptual/release-notes-azure-cli.md]: Requires issue link

File: docs-ref-conceptual/release-notes-azure-cli.md [hunk 8]
```diff
@@ -14,6 +14,93 @@ ms.service: multiple
...
+
+* [PREVIEW] Changed to use the API 2017-12-01-preview
+
+### Service Bus
+
+* Initial release
+
+### Storage
+
+* Fixed issue with missing endpoint suffix in `storage blob copy` [#] [?]
```
Comment (sptramer) [inline docs-ref-conceptual/release-notes-azure-cli.md]: Requires issue link, and not sure what command it refers to. Could not find a blob copy command.

**Comment by derekbekoe** [comment]
@sptramer We haven't been requiring links to issues in all other releases. Often times, the changes don't have an associated GitHub issue or it's difficult to parse through the commits to find the GitHub issue. Why are we starting to do it now?

**Comment by sptramer** [comment]
If there is an associated github issue, it should be linked in the changelog. This is standard practice for open source products which fix issues reported through github.

File: docs-ref-conceptual/release-notes-azure-cli.md (hunk 2 above)
Comment (derekbekoe) [reply docs-ref-conceptual/release-notes-azure-cli.md]: @williexu Can you provide the issue link?

File: docs-ref-conceptual/release-notes-azure-cli.md (hunk 3 above)
Comment (derekbekoe) [reply docs-ref-conceptual/release-notes-azure-cli.md]: @williexu Can you provide the issue link?

File: docs-ref-conceptual/release-notes-azure-cli.md (hunk 1 above)
Comment (derekbekoe) [reply docs-ref-conceptual/release-notes-azure-cli.md]: I think 'author' should really be just 'add' which would make it clearer.

File: docs-ref-conceptual/release-notes-azure-cli.md (hunk 4 above)
Comment (derekbekoe) [reply docs-ref-conceptual/release-notes-azure-cli.md]: @williexu?

File: docs-ref-conceptual/release-notes-azure-cli.md (hunk 6 above)
Comment (derekbekoe) [reply docs-ref-conceptual/release-notes-azure-cli.md]: @tjprescott?

File: docs-ref-conceptual/release-notes-azure-cli.md (hunk 5 above)
Comment (derekbekoe) [reply docs-ref-conceptual/release-notes-azure-cli.md]: @tjprescott?

File: docs-ref-conceptual/release-notes-azure-cli.md (hunk 8 above)
Comment (derekbekoe) [reply docs-ref-conceptual/release-notes-azure-cli.md]: @williexu @troydai?

File: docs-ref-conceptual/release-notes-azure-cli.md (hunk 7 above)
Comment (derekbekoe) [reply docs-ref-conceptual/release-notes-azure-cli.md]: @williexu @troydai?

File: docs-ref-conceptual/release-notes-azure-cli.md (hunk 8 above)
Comment (troydai) [reply docs-ref-conceptual/release-notes-azure-cli.md]: https://github.com/Azure/azure-cli/issues/4971

File: docs-ref-conceptual/release-notes-azure-cli.md (hunk 5 above)
Comment (tjprescott) [reply docs-ref-conceptual/release-notes-azure-cli.md]: `traffic-manager profile update`

File: docs-ref-conceptual/release-notes-azure-cli.md (hunk 6 above)
Comment (tjprescott) [reply docs-ref-conceptual/release-notes-azure-cli.md]: I don't think there was a specific issue posted for this.

File: docs-ref-conceptual/release-notes-azure-cli.md (hunk 2 above)
Comment (williexu) [reply docs-ref-conceptual/release-notes-azure-cli.md]: Persistence: https://github.com/Azure/azure-cli/issues/5625
History with scope: https://github.com/Azure/azure-cli/issues/3016

File: docs-ref-conceptual/release-notes-azure-cli.md (hunk 3 above)
Comment (williexu) [reply docs-ref-conceptual/release-notes-azure-cli.md]: There's no issue for this, I just made a pr that included this change. In interactive, you can use `--progress` to start a thread with the operation, and a progress meter will appear. This was broken.

File: docs-ref-conceptual/release-notes-azure-cli.md (hunk 4 above)
Comment (williexu) [reply docs-ref-conceptual/release-notes-azure-cli.md]: When interactive mode starts up, a command table is loaded and there may be errors  (especially when using the stack profile) such as missing parameters.
This causes the table loading and the completers (which rely on the table) to fail.
https://github.com/Azure/azure-cli/issues/5688

File: docs-ref-conceptual/release-notes-azure-cli.md (hunk 7 above)
Comment (derekbekoe) [reply docs-ref-conceptual/release-notes-azure-cli.md]: https://github.com/Azure/azure-cli/issues/5286

**Comment by derekbekoe** [comment]
@sptramer All questions answered. Can you take another look?

**Comment by formulahendry** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs-ref-conceptual/release-notes-azure-cli.md](https://github.com/derekbekoe/azure-docs-cli-python/blob/az-2.0.29/docs-ref-conceptual/release-notes-azure-cli.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/cli/azure/release-notes-azure-cli?view=azure-cli-latest&branch=pr-en-us-683) (azure-cli-latest)<br/> [View](https://review.docs.microsoft.com/en-us/cli/azure/release-notes-azure-cli?view=azure-cli-2017-03-09-profile&branch=pr-en-us-683) (azure-cli-2017-03-09-profile) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C3%5C15%5C0c435b1f-40ec-55c2-6a92-4ae06d82d844%5CPullRequest%5C201803151716007751-683%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=z1Ffw4L7lVUekKsS1WNaI%2Ffw8Iq7oPnMdCT%2FMna3dQI%3D&st=2018-03-15T17%3A15%3A13Z&se=2018-04-15T17%3A20%3A13Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by formulahendry** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs-ref-conceptual/release-notes-azure-cli.md](https://github.com/derekbekoe/azure-docs-cli-python/blob/az-2.0.29/docs-ref-conceptual/release-notes-azure-cli.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/cli/azure/release-notes-azure-cli?view=azure-cli-latest&branch=pr-en-us-683) (azure-cli-latest)<br/> [View](https://review.docs.microsoft.com/en-us/cli/azure/release-notes-azure-cli?view=azure-cli-2017-03-09-profile&branch=pr-en-us-683) (azure-cli-2017-03-09-profile) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C3%5C15%5C0c435b1f-40ec-55c2-6a92-4ae06d82d844%5CPullRequest%5C201803152316493052-683%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=u48cu32UK0gdOv3IjtdPDNtr8hVAHE4LE8slWq4nvDs%3D&st=2018-03-15T23%3A15%3A06Z&se=2018-04-15T23%3A20%3A06Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Pull request merged**